### PR TITLE
👽 Support sign with likeWallet

### PR DESCRIPTION
**Pull request by AuroraHuang22**
- Remote config changes: 
> "COSMOS_CHAIN_ID": "likecoin-public-testnet-5",
"COSMOS_LCD_URL": "https://node.testnet.like.co/rpc/",
"COSMOS_FRACTION_DENOM": "nanoekil",
"CURRENT_PREFIX": "like",

### CODE REVIEW

File: app/models/transfer-store/transfer-store.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export const TransferStoreModel = TxStoreModel
   }))
   .views(self => ({
     get receiverAddress() {
-      return self.liker ? self.liker.cosmosWallet : self.target
+      return self.liker ? self.liker.likeWallet : self.target
```
Comment (williamchong) [inline app/models/transfer-store/transfer-store.ts:23]: Would this PR release before or after chain upgrade? We do not support like prefix before upgrade is complete

File: app/services/authcore/authcore.ts [hunk 2]
```diff
@@ -235,6 +222,27 @@ export class AuthCoreAPI {
       hrp
     )
 
+    return [bech32Address]
```
Comment (nwingt) [inline app/services/authcore/authcore.ts:225]: Seem no need to return the address in array

File: app/services/authcore/authcore.ts [hunk 3]
```diff
@@ -235,6 +222,27 @@ export class AuthCoreAPI {
...
+  }
+
+  async getCosmosAddressesAndPubKeys() {
+    const pubKeys = await this.getPubKeys()
+    const addresses = this.getBech32Address(this.cosmosChainId,pubKeys[0])
+    return { addresses, pubKeys }
+  }
```
Comment (nwingt) [inline app/services/authcore/authcore.ts:232]: Should create each address with looping of the pubKeys

File: app/models/transfer-store/transfer-store.ts (hunk 1 above)
Comment (nwingt) [reply app/models/transfer-store/transfer-store.ts:23]: 1. keep cosmosWallet for now, and release another version after the chain upgrade
2. Config which addresses to use with remote config

Comment (williamchong) [reply app/models/transfer-store/transfer-store.ts:23]: both would work, but for 2 the chain id won't change, so maybe need a prefix field

Comment (AuroraHuang22) [reply app/models/transfer-store/transfer-store.ts]: Added a remote config `CURRENT_PREFIX": "like`,  now can switch this config to return a different address.

File: .env [hunk 4]
```diff
@@ -41,6 +41,7 @@ COSMOS_DENOM=LIKE
 COSMOS_FRACTION_DENOM=nanolike
 COSMOS_FRACTION_DIGITS=9
 COSMOS_GAS_PRICE=0
+CURRENT_PREFIX=
```
Comment (nwingt) [inline .env:44]: I suggest naming starts with `COSMOS_` for Cosmos related config
```suggestion
COSMOS_ADDRESS_PREFIX=
```

File: app/models/transfer-store/transfer-store.ts [hunk 5]
```diff
@@ -20,7 +20,18 @@ export const TransferStoreModel = TxStoreModel
...
-      return self.liker ? self.liker.likeWallet : self.target
+      const currentPrefix = self.env.appConfig.getValue("CURRENT_PREFIX")
+      if (self.liker) {
+        switch (currentPrefix) {
+          case "cosmos":
+            return self.liker.cosmosWallet
+          case "like":
+            return self.liker.likeWallet
+          default:
+            return self.liker.cosmosWallet
```
Comment (nwingt) [inline app/models/transfer-store/transfer-store.ts:31]: More neaty
```suggestion
          case "like":
            return self.liker.likeWallet

          case "cosmos":
          default:
            return self.liker.cosmosWallet
```

**Review by nwingt — APPROVED** [review]

**Pull request merged**